lib/compat/wordpress-6.1/script-loader.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Load global styles assets in the front-end.
+ *
+ * @package gutenberg
+ */
+
+function filter_out_block_nodes( $nodes ) {
```

Comment:
If this is merged we won't need this anymore, right?

Reply:
we need it in a filter for the case where loading separate assets is disabled.

Reply:
So the script loader filters the output of the class that handles "processing of structures that adhere to the theme.json spec". Why not instantiate with a config?

Reply:
The want the WP_Theme_Json class to be ignorant of the implementation details of how the CSS is being used - so passing a config which determines whether or not to include blocks goes against that principle.

There's more context here: https://github.com/WordPress/gutenberg/pull/40513#pullrequestreview-954837135

Reply:
Let's leave a note in the code here so that it's immediately clear to future readers.